repository/src/main/java/org/alfresco/repo/action/executer/NodeSizeDetailActionExecutor.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.action.executer;
+
+import org.alfresco.repo.action.ParameterizedItemAbstractBase;
+import org.alfresco.repo.cache.SimpleCache;
+import org.alfresco.service.cmr.action.Action;
+import org.alfresco.service.cmr.action.ParameterDefinition;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.cmr.search.ResultSet;
+import org.alfresco.service.cmr.search.SearchParameters;
+import org.alfresco.service.cmr.search.SearchService;
+import org.alfresco.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *   NodeSizeDetailActionExecutor
+ *   Executing Alfresco FTS Query to find size of Folder Node
+ */
+
+public class NodeSizeDetailActionExecutor extends ActionExecuterAbstractBase
+{
```

Comment:
Please find the updated inline status for the above points:

1. Instead of using the Action Executor approach for asynchronous, now we are utilizing ThreadpoolExecutor with elastic search configuration.
2. As suggested, we have changed the flow of SimpleCache.
3. As suggested, we have updated.
4. We are logging search service errors rather than keeping them in the API response. The errors are logged in the backend system and will be extracted to the log extractor whatever client will be using. For the enduser he will simply see that the operation is completed.